### PR TITLE
feat: Implement comprehensive stale evaluation handling

### PR DESCRIPTION
**Pull request by OAGr**
## Summary
Implements complete stale evaluation handling system in three phases:

### Phase 1: Server-side filtering for stale evaluations
- Modified `DocumentModel.getDocumentWithEvaluations()` to filter out evaluations from older document versions
- Added `includeStale` parameter for views that need all evaluations (history, management)
- Prevents broken highlights in reader view by hiding stale evaluations at database level

### Phase 2: Auto re-evaluation with cost warnings
- Added warning dialogs for document edit and re-upload actions
- Shows evaluation count and cost implications before making changes
- Automatically queues re-evaluations for all existing evaluations on document update
- Minimizes staleness window to just a few minutes

### Phase 3: UI enhancements for version awareness
- Added version mismatch indicators in evaluation history views
- Shows document version (Doc vX) for each evaluation
- Created reusable `StaleBadge` and `WarningDialog` components

### Refactoring: Server-side staleness calculation
- Moved staleness calculation from client-side to server-side
- Added `isStale` boolean field to evaluation and version schemas
- Simplified client components to use server-provided boolean
- Better performance and consistency across components

## Key Benefits
- ✅ **No more broken highlights** - Stale evaluations hidden in reader view
- ✅ **Cost transparency** - Clear warnings before re-evaluation triggers
- ✅ **Minimal staleness** - Auto re-evaluation keeps evaluations current
- ✅ **Historical access** - All evaluation versions preserved and accessible
- ✅ **Performance** - Server-side filtering reduces client computation
- ✅ **Consistency** - Single source of truth for staleness logic

## Technical Implementation
- **Database level filtering** prevents stale data from reaching client
- **Automatic job queueing** ensures evaluations stay current
- **Comprehensive error handling** for edge cases in version matching
- **Extensive unit tests** for filtering logic and DocumentModel methods
- **Reusable components** following DRY principles

## Test Coverage
- Unit tests for DocumentModel filtering logic
- Integration tests for update workflow
- Edge case handling (no versions, invalid data, etc.)
- TypeScript type safety maintained throughout

## Files Changed
- `src/models/Document.ts` - Core filtering and auto re-evaluation logic  
- `src/types/documentSchema.ts` - Added `isStale` field to schemas
- `src/components/` - New reusable `StaleBadge` and `WarningDialog` components
- Multiple client components simplified to use server-side `isStale` boolean
- Comprehensive test suite added

This implementation solves the core UX problem of broken highlights while providing clear cost transparency and maintaining access to historical evaluations.

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: manual_migration.sql [hunk 1]
```diff
@@ -0,0 +1,10 @@
+-- Add isStale column to EvaluationVersion table
```
Comment (OAGr) [inline manual_migration.sql:1]: we don't need this manual_migration anymore, please delete

File: src/app/docs/[docId]/evaluations/EvaluationsClient.tsx [hunk 2]
```diff
@@ -40,6 +40,18 @@ export default function EvaluationsClient({
   const [agentsWithEvaluations, setAgentsWithEvaluations] = useState<
     AgentWithEvaluation[]
   >([]);
+
+  // Get current document version from evaluation data
+  const currentDocumentVersion = React.useMemo(() => {
```
Comment (OAGr) [inline src/app/docs/[docId]/evaluations/EvaluationsClient.tsx:45]: why are we doing this? At least, can we do this in the server layer? Like, "isCurrent"? 

Do you think this should be another db field?
